cmd/uid_provider/api/handlers/handler.go

```diff
@@ -0,0 +1,128 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"gitlab.inspr.dev/inspr/core/cmd/uid_provider/client"
+	"gitlab.inspr.dev/inspr/core/pkg/rest"
+)
+
+// Handler is a structure which cointains methods to handle
+// requests received by the UID Provider API
+type Handler struct {
+	rdb client.RedisManager
+	ctx context.Context
+}
+
+// NewHandler instantiates a new Handler structure
+func NewHandler(rdb client.RedisManager) *Handler {
+	return &Handler{
+		rdb: rdb,
+	}
+}
+
+// CreateUserHandler handles user creation requests
+func (h *Handler) CreateUserHandler(rw http.ResponseWriter, r *http.Request) {
+	type ReceivedDataCreate struct {
+		UID string
+		Pwd string
+		Usr client.User
+	}
```

Comment:
annotate the struct fields to contain the correct tags for the json serializer, just so that it is clear what the naming for each field is